juju/juju
# Uniter reacts to relation suspended flag; new suspending/joining states

## Description of change

The uniter now listens to the relation suspended flag to determine if a relation needs to be suspended, rather than status itself. Two new status values are introduced - "joining" and "suspending". When the CLI updates a relation status flag, the status changes to one of the above values; when the uniter has processed the change, it then sets "suspended" or "joined" accordingly. A relation status also starts out as "joining" until the uniter first runs the necessary hooks.

## QA steps

Bootstrap a CMR scenario, mediawiki related to mysql in another model.
watch juju status on the mediawiki model
in the mysql model:
suspend the remote relation
observe the status change in the watch status to suspending and then suspended, relation becomes broken
resume the remote relation
observe the status change in the watch status to joining and then joined, relation becomes ok again
for both the above, listing offers in mysql model reflects the correct status
